gitlab/resource_gitlab_project_access_token.go

```diff
@@ -0,0 +1,180 @@
+				Computed: true,
+			},
+			"user_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceGitlabProjectAccessTokenCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	project := d.Get("project").(int)
+	options := &gitlab.CreateProjectAccessTokenOptions{
+		Name:   gitlab.String(d.Get("name").(string)),
+		Scopes: *stringSetToStringSlice(d.Get("scopes").(*schema.Set)),
+	}
+
+	log.Printf("[DEBUG] create gitlab ProjectAccessToken %s %s", *options.Name, options.Scopes)
+
+	if v, ok := d.GetOk("expires_at"); ok {
+		parsedExpiresAt, err := time.Parse("2006-01-02", v.(string))
+		if err != nil {
+			return fmt.Errorf("Invalid expires_at date: %v", err)
+		}
+		parsedExpiresAtISOTime := gitlab.ISOTime(parsedExpiresAt)
+		options.ExpiresAt = &parsedExpiresAtISOTime
+		log.Printf("[DEBUG] create gitlab ProjectAccessToken with expires_at %s", *options.ExpiresAt)
+	}
+
+	projectAccessToken, _, err := client.ProjectAccessTokens.CreateProjectAccessToken(project, options)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(strconv.Itoa(projectAccessToken.ID))
+	d.Set("token", projectAccessToken.Token)
+
+	return resourceGitlabProjectAccessTokenRead(d, meta)
+}
+
+func resourceGitlabProjectAccessTokenRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	project := d.Get("project").(int)
+	projectAccessTokenID, err := strconv.Atoi(d.Id())
+
+	if err != nil {
+		return fmt.Errorf("%s cannot be converted to int", d.Id())
+	}
+
+	log.Printf("[DEBUG] read gitlab ProjectAccessToken %d, project ID %d", projectAccessTokenID, project)
+
+	//there is a slight possibility to not find an existing item, for example
+	// 1. item is #101 (ie, in the 2nd page)
+	// 2. I load first page (ie. I don't find my target item)
+	// 3. A concurrent operation remove item 99 (ie, my target item shift to 1st page)
+	// 4. a concurrent operation add an item
+	// 5: I load 2nd page  (ie. I don't find my target item)
+	// 6. Total pages and total items properties are unchanged (from the perspective of the reader)
+
+	page := 1
+	for page != 0 {
+		projectAccessTokens, response, err := client.ProjectAccessTokens.ListProjectAccessTokens(project, &gitlab.ListProjectAccessTokensOptions{Page: page, PerPage: 100})
+		if err != nil {
+			return err
+		}
+
+		for _, projectAccessToken := range projectAccessTokens {
+			if projectAccessToken.ID == projectAccessTokenID {
+
+				d.Set("project", project)
+				d.Set("name", projectAccessToken.Name)
+				if projectAccessToken.ExpiresAt != nil {
+					d.Set("expires_at", projectAccessToken.ExpiresAt.String())
+				}
+				d.Set("active", projectAccessToken.Active)
+				d.Set("created_at", projectAccessToken.CreatedAt.String())
+				d.Set("revoked", projectAccessToken.Revoked)
+				d.Set("user_id", projectAccessToken.UserID)
+				d.Set("scopes", projectAccessToken.Scopes) // lintignore: R004,XR004 // TODO: Resolve this tfproviderlint issue
```

Comment:
This is likely a bug. `R004` is telling you that `projectAccessToken.Scopes` is not a valid type. What did the linter issue message say before you ignored it? See https://github.com/bflad/tfproviderlint/tree/main/passes/R004

Reply:
I checked the GitHub Actions history and it looks like there was never a `R004` issue on this line. So, disregard!

Reply:
I'm worried about this again, since I've seen there have been breaking changes in the go-gitlab package around slices. Can you resolve the `XR004` issue and remove the `//lintignore` comment? (You can resolve it by handling the error return value.)

Reply:
Modified the code to remove the XR004